enhancements/machine-config/on-cluster-builds.md

```diff
@@ -0,0 +1,1083 @@
+| `.spec.machineConfigPool`         | Required                  | Name of the MachineConfigPool to produce images for.                                                                                                                                                                                                                                                                          |
+| `.spec.renderedMachineConfig`     | Optional                  | Name of the rendered MachineConfig to bake into the image. If provided, rebuilds will not occur whenever the MachineConfigPool config changes.                                                                                                                                                                                |
+| `.spec.maxRetries`                | Optional                  | The maximum number of retries for a retriable build before a failure status is added. This will have a hard-coded default of 3 retries. See Retriable Builds section for more info on what this term means.                                                                                                                   |
+| `.spec.rolloutMode`               | Optional                  | Controls whether the built OS image is immediately rolled out to the target MachineConfigPool. Options include `immediate`, `deferred`. `immediate` will roll the image out once the build process is completed. `deferred` will require a cluster admin to manually change the MachineOSImage name on the MachineConfigPool. |
+| `.status.conditions[]`            | Required                  | Additional statuses will be added to indicate what state the build is in. Statuses will include: `BuildSuccess`, `BuildFailed`, `Building`, `BuildPending`.                                                                                                                                                                   |
+
+MachineOSBuilds can also be augmented with the following annotations:
+
+| **Annotation Name:**                             | **Description:**                                                                                                           |
+| ------------------------------------------------ | -------------------------------------------------------------------------------------------------------------------------- |
+| `machineconfiguration.openshift.io/rebuildImage` | When present, this will clear any failed image builds (and any supporting transitory objects created) and retry the build. |
+
+#### MachineOSImage
+
+MachineOSImages are a reference to the final OS image that is produced by the
```

Comment:
Why is this not just status for a build?

Strikes me that each image build is like a job, so wouldn't we expect each image to be built, on request by some object being created, and then the output being in the status?

Seems you have 2(?) objects achieving this right now?